activestorage/app/controllers/active_storage/blobs/proxy_controller.rb

```diff
@@ -6,9 +6,14 @@ class ActiveStorage::Blobs::ProxyController < ActiveStorage::BaseController
   include ActiveStorage::SetHeaders
 
   def show
-    http_cache_forever public: true do
+    if request.headers["HTTP_RANGE"].present?
       set_content_headers_from @blob
-      stream @blob
+      stream_chunk @blob
+    else
+      http_cache_forever public: true do
```

Comment:
Considering the following scenario:
- First there is a request without `HTTP_RANGE` gets cached in the browser and a CDN
- Then there is a request with `HTTP_RANGE` for the same resource.

Will the browser and CDN know that they should hit the server or will they try to serve a cached response because of the `http_cache_forever public: true`?

Reply:
I was trying to look into this but I'm a bit out of my depth. I can't find anything on how the browser will respond. I tried testing myself through safari and I either couldn't get it to cache the request, or it ignored the cache. Either way it still seemed to send out the byte range requests